### DISCUSSION TITLE
models.py (line 473) - Mixins

### DISCUSSION THREAD

**Discussion by Grok42** [reflexsoar/reflex-api#155] [Ideas] [answered]
Greetings to you both. This comment is primarily directed at Mr. Carroll. 

Recently downloaded and installed your release and was curious about the code base, so browsing the repo.  I've never used SQLAlchemy, as most of my experience is with straight SQL script or PowerShell SQL. Been considering using some Python SQL (since Py is so much prettier) and I noticed a few comments regarding Mixins that sparked interest. Figured that'd be good a place as any to learn some on SQLAlchemy.

The comments that caught my eye ( [models.py](https://github.com/reflexsoar/reflex/blob/master/reflex-api/app/models.py) ):
`473  # TODO: Figure out how to move this to a mixin, it just doesn't want to work`
 
 Found these this post while learning about Mixins.  
[SQLAlchemy generic relationship mixin -Stack Overflow](https://stackoverflow.com/questions/36815579/sqlalchemy-generic-relationship-mixin)

It mentions, that as the mixin gets created, any columns with the same name get overwritten so you're left with the last same-named column added.

Most posts and articles I browsed use the name "id" in their mixins.  I noticed your base above, here, uses "id".
`class Base(db.Model):`
`    __abstract__ = True`
`        id = db.Column(db.Integer, primary_key=True, autoincrement=True)`

If you're using "id" also in your mixins, as is used frequently from what I read, when you combine mixin with your base, could one of them be getting overwritten?  The respondent in the above forum post suggests snake-case distinct names for mixin ids. 

No idea what issue you're seeing, but it was interesting reading, taught me a bit of Python SQL, and thought I'd shoot you the forum post.

Cheers,
John

**Comment by n3tsurge** [comment, marked as answer]
We actually no longer rely on SQL alchemy in our code base this was an old comment/issue from several years back. You won't find this file in the latest main/dev branch.

Appreciate the ideas and resources tho!  I honestly can't even remember what issue I was having because I've used Mixins in other projects before.

**Comment by Grok42** [comment]
Excellent. Glad to hear you got whatever issue worked out! Guess I just latched onto the SQL Alchemy portions I saw while initially browsing. I'm moving toward more need for Python dev and do quite a bit of SQL, so it was great finding a new tool built for it.  We're nearly all Windows-based so majority of my development revolves around PowerShell.  Been working with a lot of Cloud APIs and most of their examples are in Python. Was becoming too much work to convert to a PowerShell equivalent, so began learning Python.  

Not heavy, by any means, into CyberSecurity, but noticed it was mentioned you can push other data to the system. Have a developed some Elastic solutions in the past so there's quite a bit here that interests me.  Look forward to learning more about ReflexSOAR.
